app/main.py

```
@@ -70,6 +103,16 @@ async def lifespan(_app: FastAPI):
     """FastAPI lifespan manager that handles startup and shutdown tasks."""
     logger.info("Application startup...")
 
+    # Register SIGTERM handler for graceful shutdown if enabled
+    if ENABLE_SIGTERM_HANDLER:
+        signal.signal(signal.SIGTERM, handle_sigterm)
+        logger.info(
+            f"SIGTERM handler registered (drain time: {BOT_MAX_DRAIN_SECONDS}s)"
+        )
+        # Start background task to monitor graceful shutdown
+        asyncio.create_task(graceful_shutdown_monitor())
```

Ruff (0.14.3)

Store a reference to the return value of `asyncio.create_task`

(RUF006)